plugins/workspace-minimap/src/positioned_minimap.ts

```diff
@@ -14,23 +14,131 @@ import {Minimap} from './minimap';
+      });
+      this.minimap.init();
+      this.setMinimapCss();
+      this.primaryWorkspace.resize();
+      Blockly.svgResize(this.minimap.minimapWorkspace);
+    }
+
+    /**
+     * Returns the bounding rectangle of the UI element in pixel units
+     * relative to the Blockly injection div.
+     * @returns The component’s bounding box.
+     */
+    getBoundingRectangle(): Blockly.utils.Rect {
+      return new Blockly.utils.Rect(
+          this.top, this.top + this.height,
+          this.left, this.left + this.width);
+    }
+
+    /**
+     * Positions the minimap.
+     * @param metrics The workspace metrics.
+     * @param savedPositions List of rectangles already on the workspace.
+     */
+    position(metrics: Blockly.MetricsManager.UiMetrics,
+        savedPositions: Blockly.utils.Rect[]): void {
+      const hasVerticalScrollbars = this.primaryWorkspace.scrollbar &&
+          this.primaryWorkspace.scrollbar.canScrollHorizontally();
+      const hasHorizontalScrollbars = this.primaryWorkspace.scrollbar &&
+          this.primaryWorkspace.scrollbar.canScrollVertically();
+
+      if (metrics.toolboxMetrics.position === Blockly.TOOLBOX_AT_LEFT ||
+           (this.primaryWorkspace.horizontalLayout &&
+            !this.primaryWorkspace.RTL)) {
+        // Right edge placement.
+        this.left = metrics.absoluteMetrics.left + metrics.viewMetrics.width -
+            this.width - this.margin;
+        if (hasVerticalScrollbars && !this.primaryWorkspace.RTL) {
+          this.left -= Blockly.Scrollbar.scrollbarThickness;
+        }
+      } else {
+        // Left edge placement.
+        this.left = this.margin;
+        if (hasVerticalScrollbars && this.primaryWorkspace.RTL) {
+          this.left += Blockly.Scrollbar.scrollbarThickness;
+        }
+      }
+
+      const startAtBottom =
+          metrics.toolboxMetrics.position === Blockly.TOOLBOX_AT_BOTTOM;
+      if (startAtBottom) {
+        // Bottom edge placement.
+        this.top = metrics.absoluteMetrics.top + metrics.viewMetrics.height -
+            this.height - this.margin;
+        if (hasHorizontalScrollbars) {
+          // The horizontal scrollbars are always positioned on the bottom.
+          this.top -= Blockly.Scrollbar.scrollbarThickness;
+        }
+      } else {
+        // Upper edge placement.
+        this.top = metrics.absoluteMetrics.top + this.margin;
+      }
+
+      // Check for collision and bump if needed.
+      let boundingRect = this.getBoundingRectangle();
+      for (let i = 0; i < savedPositions.length; i++) {
+        if (boundingRect.intersects(savedPositions[i])) {
+          if (startAtBottom) {
+            this.top = savedPositions[i].top - this.height - this.margin;
+          } else {
+            this.top = savedPositions[i].bottom + this.margin;
+          }
+          // Recheck other savedPositions.
+          boundingRect = this.getBoundingRectangle();
+          i = -1;
+        }
+      }
+      this.setMinimapCss();
```

Comment:
Hm you might want to trigger an `svgResize` here as well. Can you check when `position` is being called and make sure you're:
  1) Calling `svgResize` when you need to.
  2) Not calling `svgResize` when you don't.

Reply:
Ok, so position is called every time there's a window resize. That said, I think you can just embed the minimap svgResize into the this.setMinimapCss. This refactors the code and also makes it so that it is only being called when necessary.